File: 프로그래머스/level2_숫자의 표현.py
def solution(n):
    answer = 1
    for i in range(1, n // 2 + 1):
        temp = i
        for j in range(i + 1, n):
            temp += j
            if temp == n:
                answer += 1
                break
            elif temp > n:
                break

    return answer

print(solution(15))

File: 프로그래머스 복습(java)/src/level2/level2_숫자의_표현.java
package level2;

public class level2_숫자의_표현 {
    static class Solution {
        public int solution(int n) {
            int answer = 1;
            for (int i = 1; i < (int) (n / 2) + 1; i++) {
                int x = 0;
                for (int j = i; j < n; j++) {
                    x += j;
                    if (x == n) {
                        answer++;
                        break;
                    } else if (x > n) {
                        break;
                    }
                }
            }
            return answer;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution(15));
    }
}
